libraries/AP_NavEKF2/AP_NavEKF2_MagFusion.cpp

```diff
@@ -272,16 +272,12 @@ void NavEKF2_core::SelectMagFusion()
                 (frontend->_mag_ef_limit > 0 && have_table_earth_field)) {
                 FuseDeclination(0.34f);
             }
-            // fuse the three magnetometer componenents sequentially
-            for (mag_state.obsIndex = 0; mag_state.obsIndex <= 2; mag_state.obsIndex++) {
-                hal.util->perf_begin(_perf_test[0]);
-                FuseMagnetometer();
-                hal.util->perf_end(_perf_test[0]);
-                // don't continue fusion if unhealthy
-                if (!magHealth) {
-                    break;
-                }
-            }
+
+            // fuse the three magnetometer componenents using sequential fusion of each axis
```

Comment:
```suggestion
            // fuse the three magnetometer components using sequential fusion of each axis
```

... or just remove the comment entirely as it now assumes it knows what that function does, meaning it can get out-of-date.